### PR TITLE
[QMMP-1291]-[Scheduler]-[CSR 4740808-1] Tasks sometimes can't be scheduled

### PR DESCRIPTION
**Pull request by VolodymyrLavrenchuk**

### CODE REVIEW

**Comment by VolodymyrLavrenchuk** [comment]
Unit tests:
изображение

**Comment by VolodymyrLavrenchuk** [comment]
Fit tests:
изображение

File: src/logger.js [hunk 1]
```diff
@@ -2,7 +2,20 @@
...
+module.exports = createLogger({
+  level: 'info',
+  format: combine(
+    splat(),
+    colorize(),
+    timestamp(),
+    myFormat
+  ),
+  transports: [new transports.Console()]
+});
```
Comment (AntonTimiskov) [inline src/logger.js:21]: Is it producing much more logs that can be sensitive?

Comment (AntonTimiskov) [reply src/logger.js:21]: Please check the log behavior.

File: src/rest-api.js [hunk 2]
```diff
@@ -83,12 +83,53 @@ const jobName = (labels) => {
...
+  } else
+  return [[name, `Z${value}Z`]]
+}
+const cleanLabel = (label) =>  label.replace(/^Z|Z$/g, '')
+const sanitizeLabels = (labels) => _.object(_.flatten(_.map(_.keys(labels), (key) => splitLongLabel(key, labels[key])), true))
+const cleanLabels = (labels) => _.object(_.map(_.keys(labels), (key) => [key, cleanLabel(labels[key])]))
+
+const combineLongLabels = (labels) => {
+	let processedKeys = []
+	return _.object(_.map(_.keys(labels), (key) => {
```
Comment (AntonTimiskov) [inline src/rest-api.js:102]: Should you add 'numeric suffix sort' for labels here before the combining operation? To have a predictable order of substrings? or it is done by something else?

File: src/logger.js (hunk 1 above)
Comment (VolodymyrLavrenchuk) [reply src/logger.js:21]: This change was done only to support breaking changes in new winston logger version.
Previous implementation worked the same:
```winston.handleExceptions(new winston.transports.Console({ colorize: true }))```

Comment (VolodymyrLavrenchuk) [reply src/logger.js:21]: Logs contains job name, labels and other properties (but not the task properties):
```
{"kind":"CronJob","metadata":{"name":"d438967c-4a5a-409d-a448-106e026e2d3a","labels":{"id0":"Zdmp-ea9ee84b-4a90-4a42-9f57-aa45c4e15f64-tasks-tZ","id1":"Zask_schedule1Z","tenant":"Zea9ee84b-4a90-4a42-9f57-aa45c4e15f64Z","type":"Zdmp-schedulingZ","taskId":"Ztask_schedule1Z","recurrence":"ZfalseZ"},"finalizers":["foregroundDeletion"]},"spec":{"schedule":"0 12 20 11 ?","successfulJobsHistoryLimit":1,"failedJobsHistoryLimit":1,"jobTemplate":{"spec":{"template":{"spec":{"enableServiceLinks":false,"containers":[{"name":"executor","image":"mmp2devwestus2registry.azurecr.io/dmp-scheduler-executor@sha256:25cef6769334430fe40270e48078835e1c688b2fa5df94007810bc85a1d7ecca","args":[],"imagePullPolicy":"IfNotPresent","env":[{"name":"JOB","value":"{\"name\":\"dmp-ea9ee84b-4a90-4a42-9f57-aa45c4e15f64-tasks-task_schedule1\",\"type\":\"rest\",\"namespace\":\"vlavrenc\",\"request\":{\"uri\":\"http://vlavrenc-internal-proxy/api/tasks/task_schedule1\",\"method\":\"PATCH\",\"json\":true,\"headers\":{\"Tenant\":\"ea9ee84b-4a90-4a42-9f57-aa45c4e15f64\"},\"body\":{\"action\":\"Start\",\"status\":\"In Progress\",\"lastResultDescription\":\"Start on schedule\"}},\"recur\":{\"start\":null,\"triggers\":[\"0 12 20 11 ?\"],\"end\":null},\"labels\":{\"tenant\":\"ea9ee84b-4a90-4a42-9f57-aa45c4e15f64\",\"type\":\"dmp-scheduling\",\"taskId\":\"task_schedule1\",\"recurrence\":\"false\"}}"}]}],"restartPolicy":"OnFailure"}}}}}}
```

File: src/rest-api.js (hunk 2 above)
Comment (VolodymyrLavrenchuk) [reply src/rest-api.js:102]: JS does have a guaranteed property order that was standardised in ES2015.
Before that it was a pseudo-standard among most engines.
https://tc39.es/ecma262/#sec-ordinaryownpropertykeys

I have a test for this behaivour.

**Pull request merged**